stl/src/stacktrace.cpp

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#include <yvals.h>
+
+#include <cstdio>
+#include <cstdlib>
+
+// clang-format off
+#include <initguid.h> // should be before any header that includes <guiddef.h>
+#include <DbgEng.h>
+#include <Shlwapi.h>
+// clang-format on
+
+#pragma comment(lib, "DbgEng.lib")
+#pragma comment(lib, "Shlwapi.lib")
+
+namespace {
+    // The below function pointer types must be in sync with <stacktrace>
+
+    using _Stacktrace_string_fill_callback = size_t(__stdcall*)(char* _Data, size_t _Size, void* _Context);

```

Comment:
It would be possible, but it would have a small impact on user throughput (as `<stacktrace>` would then need to include a subheader). That seems excessive for two typedefs (whereas `xfilesystem_abi.h` contains a ton of structures and constants).

Some precedent - we've been willing to ship a small amount of duplication with comments that `inc` and `src` must match, e.g. in the `shared_mutex` implementation.

https://github.com/microsoft/STL/blob/60decd0d829e68666b556780002c83605d748d80/stl/inc/xthreads.h#L91-L99
https://github.com/microsoft/STL/blob/60decd0d829e68666b556780002c83605d748d80/stl/src/sharedmutex.cpp#L11-L20

My preference would be to leave this as-is, but if you feel strongly that we should avoid duplication going forward, I would not object to adding a subheader (the throughput impact is admittedly small, and a non-issue for Standard Library Modules which are the long-term future).